Test failure on main with dask >= 2021.4

```
❯ pytest dask_jobqueue/tests/test_slurm.py -k job_script
========================================= test session starts ==========================================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/lesteve/dev/dask-jobqueue
plugins: asyncio-0.15.1, cov-2.12.1
collected 6 items / 5 deselected / 1 selected                                                          

dask_jobqueue/tests/test_slurm.py F                                                              [100%]

=============================================== FAILURES ===============================================
___________________________________________ test_job_script ____________________________________________

    def test_job_script():
        with SLURMCluster(
            walltime="00:02:00", processes=4, cores=8, memory="28GB"
        ) as cluster:
    
            job_script = cluster.job_script()
            assert "#SBATCH" in job_script
            assert "#SBATCH -J dask-worker" in job_script
>           assert "--memory-limit 7.00GB " in job_script
E           AssertionError: assert '--memory-limit 7.00GB ' in '#!/usr/bin/env bash\n\n#SBATCH -J dask-worker\n#SBATCH -n 1\n#SBATCH --cpus-per-task=8\n#SBATCH --mem=27G\n#SBATCH -t...:42463 --nthreads 2 --nprocs 4 --memory-limit 6.52GiB --name dummy-name --nanny --death-timeout 60 --protocol tcp://\n'

dask_jobqueue/tests/test_slurm.py:63: AssertionError
======================================= short test summary info ========================================
FAILED dask_jobqueue/tests/test_slurm.py::test_job_script - AssertionError: assert '--memory-limit 7....
=================================== 1 failed, 5 deselected in 1.53s ====================================
```

This seems like the memory is in GiB rather than GB, maybe a change in dask/distributed?
- it passes on the CI (dask 2021.3 and distributed 2021.3 see [this](https://github.com/dask/dask-jobqueue/runs/3130790721#step:5:47)) which I can reproduce locally. Side-comment: why are we using dask 2021.3? Probably because we test with python 3.6.
- it fails with dask 2021.4 and distributed 2021.4

Needs further investigation ...

--- a/dask_jobqueue/tests/test_htcondor.py
+++ b/dask_jobqueue/tests/test_htcondor.py
@@ -5,6 +5,7 @@ import pytest
 from distributed import Client
 
 import dask
+from dask.utils import format_bytes, parse_bytes
 
 from dask_jobqueue import HTCondorCluster
 
@@ -44,7 +45,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--memory-limit 50.00MB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("50MB")).replace(" ", "")
+        assert f"--memory-limit {formatted_bytes}" in job_script
         assert "--nthreads 2" in job_script
         assert "--nprocs 2" in job_script
 

--- a/dask_jobqueue/tests/test_lsf.py
+++ b/dask_jobqueue/tests/test_lsf.py
@@ -8,7 +8,7 @@ from time import sleep, time
 import dask
 import pytest
 from dask.distributed import Client
-from distributed.utils import parse_bytes
+from dask.utils import format_bytes, parse_bytes
 
 from dask_jobqueue import LSFCluster, lsf
 
@@ -100,7 +100,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
     with LSFCluster(
         queue="general",
@@ -126,7 +127,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
     with LSFCluster(
         walltime="1:00",
@@ -149,7 +151,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 1 --memory-limit 16.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("16GB")).replace(" ", "")
+        assert f"--nthreads 1 --memory-limit {formatted_bytes}" in job_script
 
 
 @pytest.mark.env("lsf")

--- a/dask_jobqueue/tests/test_oar.py
+++ b/dask_jobqueue/tests/test_oar.py
@@ -1,7 +1,9 @@
 import sys
 
-from dask_jobqueue import OARCluster
 import dask
+from dask.utils import format_bytes, parse_bytes
+
+from dask_jobqueue import OARCluster
 
 
 def test_header():
@@ -40,7 +42,8 @@ def test_job_script():
         job_script = cluster.job_script()
         assert "#OAR" in job_script
         assert "#OAR -n dask-worker" in job_script
-        assert "--memory-limit 7.00GB " in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--memory-limit {formatted_bytes}" in job_script
         assert "#OAR -l /nodes=1/core=8,walltime=00:02:00" in job_script
         assert "#OAR --project" not in job_script
         assert "#OAR -q" not in job_script
@@ -51,7 +54,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
     with OARCluster(
         walltime="00:02:00",
@@ -67,7 +71,8 @@ def test_job_script():
         job_script = cluster.job_script()
         assert "#OAR" in job_script
         assert "#OAR -n dask-worker" in job_script
-        assert "--memory-limit 7.00GB " in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--memory-limit {formatted_bytes}" in job_script
         assert "#OAR -l /nodes=1/core=8,walltime=00:02:00" in job_script
         assert "#OAR --project" not in job_script
         assert "#OAR -q" not in job_script
@@ -80,7 +85,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
 
 def test_config_name_oar_takes_custom_config():

--- a/dask_jobqueue/tests/test_pbs.py
+++ b/dask_jobqueue/tests/test_pbs.py
@@ -1,8 +1,11 @@
 import sys
 from time import sleep, time
 
-import dask
 import pytest
+
+import dask
+from dask.utils import format_bytes, parse_bytes
+
 from dask.distributed import Client
 
 from dask_jobqueue import MoabCluster, PBSCluster
@@ -72,7 +75,8 @@ def test_job_script(Cluster):
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
     with Cluster(
         queue="regular",
@@ -95,7 +99,8 @@ def test_job_script(Cluster):
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
 
 @pytest.mark.env("pbs")

--- a/dask_jobqueue/tests/test_sge.py
+++ b/dask_jobqueue/tests/test_sge.py
@@ -5,6 +5,7 @@ from distributed import Client
 
 from dask_jobqueue import SGECluster
 import dask
+from dask.utils import format_bytes, parse_bytes
 
 from . import QUEUE_WAIT
 
@@ -83,10 +84,12 @@ def test_job_script(tmpdir):
         resource_spec="h_vmem=12G,mem_req=12G",
     ) as cluster:
         job_script = cluster.job_script()
+        formatted_bytes = format_bytes(parse_bytes("6GB")).replace(" ", "")
+
         for each in [
             "--nprocs 2",
             "--nthreads 3",
-            "--memory-limit 6.00GB",
+            f"--memory-limit {formatted_bytes}",
             "-q my-queue",
             "-P my-project",
             "-l h_rt=02:00:00",

--- a/dask_jobqueue/tests/test_slurm.py
+++ b/dask_jobqueue/tests/test_slurm.py
@@ -5,6 +5,7 @@ import pytest
 from distributed import Client
 
 import dask
+from dask.utils import format_bytes, parse_bytes
 
 from dask_jobqueue import SLURMCluster
 
@@ -60,7 +61,8 @@ def test_job_script():
         job_script = cluster.job_script()
         assert "#SBATCH" in job_script
         assert "#SBATCH -J dask-worker" in job_script
-        assert "--memory-limit 7.00GB " in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--memory-limit {formatted_bytes}" in job_script
         assert "#SBATCH -n 1" in job_script
         assert "#SBATCH --cpus-per-task=8" in job_script
         assert "#SBATCH --mem=27G" in job_script
@@ -74,7 +76,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
     with SLURMCluster(
         walltime="00:02:00",
@@ -105,7 +108,8 @@ def test_job_script():
             "{} -m distributed.cli.dask_worker tcp://".format(sys.executable)
             in job_script
         )
-        assert "--nthreads 2 --nprocs 4 --memory-limit 7.00GB" in job_script
+        formatted_bytes = format_bytes(parse_bytes("7GB")).replace(" ", "")
+        assert f"--nthreads 2 --nprocs 4 --memory-limit {formatted_bytes}" in job_script
 
 
 @pytest.mark.env("slurm")

